### PR TITLE
feat: Free drops requires activated email

### LINKED ISSUE #8575: Free drops requires activated email
Seems free drops are being ineffective currently and looking way to keep people real.

We can authenticate against active email. 
We will check **if the entered email is activated**, and then the email can proceed to claim free drop. 

@exezbcz can drop design

Checking, can we mark at the user device that inputted email is activated at least in local storage?
I am not sure if it was solved here
- https://github.com/kodadot/nft-gallery/issues/8485

--------------

# Design 

helo helo 

- simple modal here, will be applied for every drop that has the email signup - that means free drops 
- professional illustrations lol

image

image

will explain design first 
- option to resend the confirmation mail
- option to change email - sends the user to the email signup (step before) 
- When clicking I have confirmed, it will check on the backend if the user confirmed, till then do nothing 
   - if yes, continue to claim modal without any other action
   - if no, display that its not confirmed, check again once the user clicks "check again" 
- when checking active, change the button state to loading state with button copy checking 

image

file: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1458-19091&mode=design&t=KV5clMMiVww9HCmP-11

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Requires https://github.com/kodadot/workers/pull/229
- [x] Requires https://github.com/kodadot/workers/pull/234
- [x] Closes #8575

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

https://github.com/kodadot/nft-gallery/assets/44554284/897b6942-a6be-4d7b-9fa3-dd3f89f42126

## Copilot Summary
copilot:summary

copilot:poem

### CODE REVIEW

**Review by Jarsen136 — APPROVED** [review]

**Review by roiLeo — COMMENTED** [review]
dark-mode svg?

**Comment by hassnian** [comment]
@prury pls test the following scenarios
- current user has no email subscription registered -> first will see the enter email 
- user has already **subscribed**  -> user could've entered their email here
CleanShot 2023-12-20 at 15 32 44@2x
they will see email confirmation step if the email is not confirmed , if it's , it will start minting  
- user has already **confirmed** email  , minting will start immediately

it might be hard to test , but this is how it should work 

thanks

**Comment by exezbcz** [comment]
> * user has already **subscribed**  -> user could've entered their email here

or on a previous drop

**Comment by vikiival** [comment]
Deployed enjoy

https://github.com/kodadot/workers/actions/runs/7274782427

**Comment by hassnian** [comment]
> dark-mode svg?

added
CleanShot 2023-12-20 at 16 53 57@2x

**Comment by hassnian** [comment]
> > * user has already **subscribed**  -> user could've entered their email here
> 
> 
> 
> or on a previous drop 
> 
> 
> 
> 

yes , after this pr is merged it will remember the newsletter subscription details

File: components/collection/drop/modal/ConfirmEmail.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    </NeoButton>
+  </div>
+
+  <div v-if="emailNotConfirmed" class="mt-4 text-center capitalize">
+    <p class="has-text-k-red is-size-7">{{ $t('drops.emailNotConfirmed') }}</p>
+  </div>
+</template>
+
+<script setup lang="ts">
+import ModalIdentityItem from '@/components/shared/ModalIdentityItem.vue'
```
Comment (roiLeo) [inline components/collection/drop/modal/ConfirmEmail.vue:54]: auto imported
```suggestion
```

File: components/collection/drop/modal/ConfirmEmail.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    {{ $t('drops.toConfirmYourSubscription') }}
+  </p>
+
+  <div>
+    <img :src="emailSubscriptionImage" />
+  </div>
+
+  <hr class="m-0 mb-2 mt-5" />
+
+  <div class="capitalize is-size-7 has-text-k-grey">
```
Comment (roiLeo) [inline components/collection/drop/modal/ConfirmEmail.vue:17]:
```suggestion
  <div class="capitalize text-xs text-k-grey">
```

File: components/collection/drop/Generative.vue [hunk 3]
```diff
@@ -404,6 +451,18 @@ const handleDropAddModalConfirm = () => {
...
 
+watch([isConfirmModalActive, emailConfirmed], ([modalActive, confirmed]) => {
+  if (modalActive && confirmed) {
+    startMinting()
+  }
+})
+
+onMounted(() => {
+  if (preferencesStore.getNewsletterSubscription.confirmed) {
+    emailConfirmed.value = true
```
Comment (roiLeo) [inline components/collection/drop/Generative.vue:462]: make this value computed instead?

File: utils/newsletter.ts [hunk 4]
```diff
@@ -28,4 +28,30 @@ async function subscribe(email: string) {
...
 
-export default { subscribe }
+type SubscriptionByEmailResponse = {
+  email: string
+  status: 'pending' | 'active'
+}
+
+export async function getSubscription(email: string) {
+  try {
+    return api<SubscriptionByEmailResponse>(`/subscribe/${email}`, {
```
Comment (roiLeo) [inline utils/newsletter.ts:38]: no await?

File: utils/newsletter.ts [hunk 5]
```diff
@@ -28,4 +28,30 @@ async function subscribe(email: string) {
...
+      method: 'GET',
+    })
+  } catch (e) {
+    throw new Error('[NEWSLETTER] Unable to get subscription' + e)
+  }
+}
+
+export async function resendConfirmationEmail(email: string) {
+  try {
+    return api('/subscribe/resend-confirmation', {
```
Comment (roiLeo) [inline utils/newsletter.ts:48]: no await?

**Review by roiLeo — APPROVED** [review]
smoll, otherwise lgtm

**Comment by yangwao** [comment]
Testing at 
- https://deploy-preview-8668--koda-canary.netlify.app/ahp/drops/chaosflakes

**Comment by yangwao** [comment]
confirmed at behiiv
image
doesn't work on this pr
image

**Comment by exezbcz** [comment]
works for me 

one thing, it would be nice to maybe send the confirmation email every time you click subscribe and claim 
- I have already inputted the address earlier but did not confirm - i had to resend the email to find it
image
image
- plus the skeleton here looks to bright

**Comment by yangwao** [comment]
> works for me

I've used something+something@gmail.com could be possible doesn't handle well?

see error - https://newsletter.w.kodadot.xyz/subscribe/something+something@gmail.com

image

**Comment by roiLeo** [comment]
> plus the skeleton here looks to bright

might be related https://github.com/kodadot/nft-gallery/pull/8541#issuecomment-1860390801

**Comment by prury** [comment]
- we need a better error message and also typo: 
image

------------

- failed sending confirmation(maybe because I'm using temporary e-mail service? or e-mail name?):
 

------------

- using another temporary service mail ✔️ 
https://github.com/kodadot/nft-gallery/assets/36627808/b6a76760-756c-447b-a952-e53ada302f23

------------

> * user has already **confirmed** the email  , minting will start immediately

**test using my personal e-mail:** my email was already confirmed because i was on beehive list already, but it asked me to confirm again, i just skipped the confirm part and was able to mint

------------


> I've used [something+something@gmail.com](mailto:something+something@gmail.com) could be possible doesn't handle well?
> 

yup, doesn't work with strange chars, nice catch! 

https://github.com/kodadot/nft-gallery/assets/36627808/5b72821e-c7e6-4a37-9ae1-fc9d118426a4

**Comment by hassnian** [comment]
> I've used [something+something@gmail.com](mailto:something+something@gmail.com) could be possible doesn't handle well?

seems like an issue on beehiiv's end , that endpoint doesn't handle emails with special chars 

CleanShot 2023-12-21 at 11 57 29@2x

CleanShot 2023-12-21 at 11 54 54@2x

**Comment by yangwao** [comment]
let's see if it build oka

**Comment by yangwao** [comment]
- https://github.com/kodadot/workers/pull/234 deployed

File: locales/en.json [hunk 6]
```diff
@@ -1475,7 +1481,18 @@
...
+    "iveConfirmed": "I've confirmed",
+    "emailNotConfirmed": "Email not confirmed. Please confirm your email address and check again",
+    "noEmail": "No email?",
+    "checkSpamFolder": "Or check your Spam folder.",
+    "wrongEmail": "Wrong email?",
+    "confirmYourSubscription": "Confirm Your Subscription",
+    "emailConfirmationSent": "Email confirmation sent",
+    "failedEmailConfirmation": "Failed sending email confirmation",
+    "failedCheckingSubscription": "Failed checking subscription",
+    "holderOfClaimed": "{0} drop already claimed with this NFT. Claim rights are not renewed after resale.",
```
Comment (yangwao) [inline locales/en.json:1495]:
```suggestion
    "holderOfClaimed": "{0} drop already claimed with this NFT. Claim rights are not renewed after resale."
```

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 7.63 USD/DOT ~ 10.485 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0x18755e46370f747497078720ac315cea96bfc627078e1a8d50184a030ae49bec](https://polkadot.subscan.io/extrinsic/0x18755e46370f747497078720ac315cea96bfc627078e1a8d50184a030ae49bec)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
green
image

**Pull request merged**